caffe2/core/operator_schema.h

```diff
@@ -371,15 +371,18 @@ class CAFFE2_API OpSchema {
     return device_inference_function_(def);
   }
 
-  // The helper is build sparse input with values, keys, and lengths; e.g.:
+  // The helper is build sparse input with values, keys, weights and lengths;
+  // e.g.:
   // values  = [1, 2, 3, 2, 4, 6, 7, 3, 6]
   // keys    = [0, 1, 4, 0, 1, 2, 5, 1, 2]
+  // weights = [1, 2, 3, 4, 5, 6, 7, 8, 9]
   //            \_____/  \________/  \__/
   // lengths =    [3,        4,       2]
-  OpSchema& ValueKeyLengthInputFillers(
+  OpSchema& WeightedValueKeyLengthInputFillers(
       size_t value_index,
       size_t key_index,
-      size_t length_index);
+      size_t length_index,
+      size_t weight_index = -1);
```

Comment:
Not a big fan of this interface....I kinda missed `-1` + `size_t` until I see `if (weight_index < fillers->size())`. Maybe just leave two separate helpers so we don't have to have an extra `if` in the implementation.

Reply:
Yeah, this diff is just testing a few things. I'd have int as input which is the case in C2.